### PR TITLE
Frontend/fix: #2546 Payment History Screen fix

### PR DESCRIPTION
**Pull request by Vignesh-772**
## Type of Change

Payment History Screen fix for showing the all transaction

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

**Review by Rohit4997 — DISMISSED** [review]

File: Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs [hunk 1]
```diff
@@ -561,12 +561,12 @@ makePaymentState state =
     title : getString GREAT_JOB,
     description : getDescription state,
     description2 : ( case getValueToLocalStore LANGUAGE_KEY of
-                          "EN_US" -> "To continue using Yatri Sathi, please complete your payment for " <> state.data.paymentState.date
-                          "HI_IN" -> "यात्री साथी का उपयोग जारी रखने के लिए, कृपया "<> state.data.paymentState.date <>" के लिए अपना भुगतान पूरा करें"
-                          "KN_IN" -> "ಯಾತ್ರಿ ಸತಿ ಬಳಸುವುದನ್ನು ಮುಂದುವರಿಸಲು, ದಯವಿಟ್ಟು "<> state.data.paymentState.date <> " ಕ್ಕೆ ನಿಮ್ಮ ಪಾವತಿಯನ್ನು ಪೂರ್ಣಗೊಳಿಸಿ"
-                          "TA_IN" -> "யாத்ரி சாத்தியைத் தொடர்ந்து பயன்படுத்த, "<> state.data.paymentState.date <> " க்கு உங்கள் கட்டணத்தைச் செலுத்தவும்"
-                          "BN_IN" -> "Yatri Sathi ব্যবহার চালিয়ে যেতে, অনুগ্রহ করে " <> state.data.paymentState.date <> " -এর জন্য আপনার অর্থপ্রদান সম্পূর্ণ করুন"
-                          _       -> "To continue using Yatri Sathi, please complete your payment for " <> state.data.paymentState.date
+                          "EN_US" -> "To continue using Yatri Sathi, please complete your payment"
```
Comment (Rohit4997) [inline Frontend/ui-driver/src/Screens/HomeScreen/ComponentConfig.purs:564]: @Vignesh-772  if variable is removed then you can add it in translations.
No need to do like this

File: Frontend/ui-common/src/Engineering/Helpers/Utils.purs [hunk 2]
```diff
@@ -51,25 +52,16 @@ foreign import loaderTextIOS :: EffectFn2 String String Unit
...
 
 loaderText :: String -> String -> Flow GlobalState Unit
-loaderText mainTxt subTxt = do
-  if os == "IOS" then do
-    _ <- liftFlow $ runEffectFn2 loaderTextIOS mainTxt subTxt
-    pure unit
-  else do
-    _ <- modifyState (\(GlobalState state) -> GlobalState state { loaderOverlay { data { title = mainTxt, subTitle = subTxt } } })
-    pure unit
+loaderText mainTxt subTxt = void $ modifyState (\(GlobalState state) -> GlobalState state { loaderOverlay { data { title = mainTxt, subTitle = subTxt } } })
```
Comment (kavya-shree-s) [inline Frontend/ui-common/src/Engineering/Helpers/Utils.purs:64]: Why is check for IOS removed ?

File: Frontend/ui-driver/src/Screens/RideHistoryScreen/Transformer.purs [hunk 3]
```diff
@@ -1,43 +1,85 @@
...
+
+type OrderWithTxn
+  = { order :: PaymentHistoryListItem.Config
+    , txns :: Array PaymentHistoryListItem.Config
+    }
+
+appendArray :: Array (Array PaymentHistoryListItem.Config) -> Array PaymentHistoryListItem.Config
+appendArray item = case (DA.head item) of
+  Nothing -> []
+  Just a ->
```
Comment (kavya-shree-s) [inline Frontend/ui-driver/src/Screens/RideHistoryScreen/Transformer.purs:46]: Use proper variables

File: Frontend/ui-driver/src/Services/Accessor.purs [hunk 4]
```diff
@@ -75,4 +75,7 @@ _certificateNumber :: forall a b c. Newtype a { certificateNumber :: b | c} => L
 _certificateNumber = lens (unwrap >>> _.certificateNumber) (\oldRec newVal -> wrap (unwrap oldRec) {certificateNumber = newVal})
 
 _distance :: forall a b c. Newtype a {distance :: c | b} => Lens' a c
-_distance = lens (unwrap >>> _.distance) (\oldRec newVal -> wrap ((unwrap oldRec) {distance = newVal}))
\ No newline at end of file
+_distance = lens (unwrap >>> _.distance) (\oldRec newVal -> wrap ((unwrap oldRec) {distance = newVal}))
+
+_orderId :: forall a b c. Newtype a { orderId :: b | c } => Lens' a b
```
Comment (kavya-shree-s) [inline Frontend/ui-driver/src/Services/Accessor.purs:80]: Remove `_orderId` as it is not used

**Review by kavya-shree-s — CHANGES_REQUESTED** [review]

File: Frontend/ui-common/src/Engineering/Helpers/Utils.purs (hunk 2 above)
Comment (Vignesh-772) [reply Frontend/ui-common/src/Engineering/Helpers/Utils.purs:64]: It is fixed no need of native loader

File: Frontend/ui-driver/spago.dhall [hunk 5]
```diff
@@ -44,7 +44,6 @@ to generate this file without the comments in this block.
   , "tracker"
   , "transformers"
   , "tuples"
-  , "lite-decode"
```
[the remark this answers is not in the document]
Comment (Vignesh-772) [reply Frontend/ui-driver/spago.dhall]: added back

**Review by MercyQueen — APPROVED** [review]

**Review by kavya-shree-s — APPROVED** [review]

**Pull request merged**